### PR TITLE
Access control: Use ResolveIdentity() for authorizing in org

### PR DESCRIPTION
**Pull request by alexanderzobnin**
**What is this feature?**

This PR removes hacky `makeTmpUser()` func for getting user identity in specific org with `ResolveIdentity()` from authn service.

**Why do we need this feature?**

We need to simplify authorization in org other than user is logged in.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/619

### CODE REVIEW

**Review by kalleep — APPROVED** [review]
LGTM, I think it would be good if @gamab also took a look at this one. He has found some fixed some issues around it in the past so just want to make sure we don't have any regressions

File: pkg/api/org_test.go [hunk 1]
```diff
@@ -5,6 +5,9 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/grafana/grafana/pkg/services/authn"
+	"github.com/grafana/grafana/pkg/services/authn/authntest"
```
Comment (IevaVasiljeva) [inline pkg/api/org_test.go:9]: Nit: import ordering

File: pkg/api/plugins_test.go [hunk 2]
```diff
@@ -12,12 +12,12 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
```
Comment (IevaVasiljeva) [inline pkg/api/plugins_test.go:20]: Nit: import ordering

File: pkg/services/accesscontrol/middleware.go [hunk 3]
```diff
@@ -194,102 +184,23 @@ func AuthorizeInOrgMiddleware(ac AccessControl, service Service, userService use
...
 
-			tmpUser, err := makeTmpUser(c.Req.Context(), service, userService, teamService, c.SignedInUser, targetOrgID)
-			if err != nil {
-				deny(c, nil, fmt.Errorf("failed to authenticate user in target org: %w", err))
-				return
+			var orgUser identity.Requester = c.SignedInUser
+			if targetOrgID != c.SignedInUser.GetOrgID() {
+				orgUser, err = authnService.ResolveIdentity(c.Req.Context(), targetOrgID, c.SignedInUser.GetID())
+				if err != nil {
+					deny(c, nil, fmt.Errorf("failed to authenticate user in target org: %w", err))
```
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/middleware.go:191]: Nit: maybe `failed to authorize user in target org` would be more accurate.

**Review by IevaVasiljeva — COMMENTED** [review]
From reading through it looks good. Only some very small nits (mostly to do with import ordering), feel free to ignore those.

I'll do some testing before approving.

**Review by IevaVasiljeva — CHANGES_REQUESTED** [review]
I just tested this for several scenarios, and it allows users with Admin rights in a single org to have Admin permissions globally. Eg, as an org Admin user I can create or edit global roles. I think it's because [FetchSyncedUserHook](https://github.com/grafana/grafana/blob/9c292d2c3f771c24e393ae08dcd5a26b5405f7c0/pkg/services/authn/authnimpl/sync/user_sync.go#L124-L135) overwrites the org ID in the identity to be the org ID of the current signed in user in case the target org ID is set to 0 (I _think_ it's because of [this conditional](https://github.com/grafana/grafana/blob/9c292d2c3f771c24e393ae08dcd5a26b5405f7c0/pkg/services/user/userimpl/store.go#L392-L395)). So when we get to evaluating permissions, we evaluate them in user's current org instead of the global org as we should.

Repro steps:
* create an org Admin user;
* try to create a global role with this org Admin user - this should fail, but it doesn't.

Other scenarios that I tested and that worked as expected:
* when _single_organization_ is set, org Admin user can update basic role;
* when _single_organization_ is set, a service account with Admin role can update basic role;
* when _single_organization_ is not set, a service account with Admin role cannot update basic role;
* user who is an Admin in one org and has lower perms in another org can't use the Admin perms in this other org;
* service account cannot do anything in an org that it doesn't belong to.

**Review by IevaVasiljeva — APPROVED** [review]
I've tested again together with changes in https://github.com/grafana/grafana/pull/85835, and I think everything works as expected.

Test scenarios covered:
* org admin or SA with org admin permissions can create a role in the org where they are an admin;
* org admin or SA with org admin permissions cannot create a role in the global org or a different org where they are not admins;
* viewer + server admin can create a role in the global org;
* org admin or SA with org admin permissions can create a role in the global org if `single_organization` is set to true;
* org admin or SA with org admin permissions can not create a role in another org where they are not admins when `single_organization` is set to true.

One scenario that is a bit weird:
* viewer + server admin can create a role in the org where the user is just a viewer. I don't remember how we treated this scenario in the past - I think this is expected 🤔

**Comment by IevaVasiljeva** [comment]
Great job on this @alexanderzobnin and @kalleep 🥇 🎉

**Pull request merged**